warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1111,6 +1111,41 @@ protected boolean timedCheckForSortedUids(QueryStopwatch timers, String stage, S
         return sortedUIDs;
     }
 
+    protected void timeScanHintRules(QueryStopwatch timers, String stage, ShardQueryConfiguration config) {
+        TraceStopwatch stopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - " + stage);
+
+        // apply runtime scan hints
+        if (config.isUseQueryTreeScanHintRules()) {
+            for (ScanHintRule<JexlNode> hintRule : config.getQueryTreeScanHintRules()) {
+                if (hintRule.getTable() != null && hintRule.apply(config.getQueryTree())) {
+                    Map<String,String> tableHints = config.getTableHints().get(hintRule.getTable());
```

Comment:
My only thought here is we are assuming that config.getTableHints is not returning a copy.  Maybe we create a local map and the config.setTableHints at the end.